weblab/entities/views.py

```diff
@@ -1004,3 +1004,63 @@ def get(self, request, *args, **kwargs):
         return JsonResponse({
             task: result
         })
+
+
+class EntityRunExperimentView(PermissionRequiredMixin, LoginRequiredMixin, EntityTypeMixin, DetailView):
+    """
+    Class for listing the possible experiment combinations
+    """
+    permission_required = 'experiments.create_experiment'
+    context_object_name = 'entity'
+    template_name = 'entities/entity_runexperiments.html'
+
+    def get_context_data(self, **kwargs):
+        entity = self.object
+
+        context = super().get_context_data(**kwargs)
+
+        # preposition to use in sentence: You may run this entity on/under the following entities
+        context['preposition'] = 'under'
+        if entity.entity_type == 'protocol':
+            context['preposition'] = 'on'
+
+        # ended up using a nested dict as nested lists caused django's unpacking in forloops to
+        # mess things up slightly
+        other_entities = Entity.objects.filter(entity_type=entity.other_type)\
+            .select_related('cachedentity').prefetch_related('cachedentity__versions', 'cachedentity__versions__tags')
+        context['object_list'] = []
+        for item in other_entities:
+            versions = item.cachedentity.versions
+            version_info = []
+            for version in versions.prefetch_related('tags'):
+                tag_list = list(version.tags.values_list('tag', flat=True))
+                commit = item.repo.get_commit(version.sha)
+                latest = item.repo.latest_commit
+                version_info.append({'commit': commit, 'tags': tag_list, 'latest': latest == commit})
+            context['object_list'].append({'id': item.name, 'versions': version_info})
+        return context
+
+    def post(self, request, *args, **kwargs):
+        experiments_to_run = request.POST.getlist('runexperimentlist[]')
+        for version in experiments_to_run:
+            ident, sha = version.split(':')
+            entity = Entity.objects.get(pk=ident)
+            commit = entity.repo.get_commit(sha)
+            record_experiments_to_run(request.user, entity, commit)
```

Comment:
You won't be able to use `record_experiments_to_run` as-is, since it is specifically for running the equivalent experiments on a new entity version that were run on its parents. But an equivalent approach of adding rows to `PlannedExperiment` should work here. In fact, here it should be easier since the POST arguments should contain (almost?) all the info you need directly, rather than needing to look at prior experiments - except for handling the re-run issue.